src/locutus/model/table.py

```diff
@@ -316,9 +316,8 @@ def get_preference(self, code=None):
         try:
             pref = self.terminology.dereference().get_preference(code=code)
 
-            # Check if pref is a dictionary and has keys with empty dictionaries as values
-            if isinstance(pref, dict) and any(isinstance(value, dict) and not value for value in pref.values()):
-                # Trigger retrieval for 'self'
+            # If code exists as a key and is empty get the Table preferences.
+            if code in pref and not pref[code]:
```

Comment:
I "think" 

if pref.get(code): 

pref.get(...) will safely return None if whatever you are asking for doesn't exist (without creating a new entry) or return the value at code if it does exist.